packages/mobx-state-tree/test/reflection.ts

```diff
@@ -0,0 +1,73 @@
+import { types, reflect } from "../src"
+import { test } from "ava"
+
+const User = types.model("User", {
+    id: types.identifier(types.string),
+    name: types.string
+})
+const Model = types
+    .model({
+        propertyName: false,
+        users: types.maybe(types.map(User)),
+        dogs: types.maybe(types.array(User)),
+        user: types.maybe(types.late(() => User))
+    })
+    .volatile(self => ({
+        volatileProperty: { propName: "halo" }
+    }))
+    .actions(self => {
+        function actionName() {
+            return 1
+        }
+        return {
+            actionName
+        }
+    })
+    .views(self => ({
+        get viewName() {
+            return 1
+        }
+    }))
+
+test("reflection model", t => {
+    const node = Model.create()
```

Comment:
can you add a test that `model({ x: types.string}` and then assert `reflect(type).properties.x === types.string`?